### PR TITLE
[IMP] project, sale_project: hootify tests

### PR DESCRIPTION
**Pull request by ppr-odoo**
task-4354290

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_project/static/tests/project_task_model.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export class ProjectMilestone extends projectModels.ProjectMilestone {
+    _name = "project.milestone";
+
+    product_uom_qty = fields.Float({ string: "Quantity" });
+    quantity_percentage = fields.Float({ string: "Percentage" });
+}
+
+Object.assign(projectModels, {
+    ProjectMilestone,
+    ProjectTask,
```
Comment (pkyriakou) [inline addons/sale_project/static/tests/project_task_model.js:21]: I don't think you have to re-assign those :thinking:

File: addons/sale_project/static/tests/project_task_list_sale_line.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { describe, expect, test } from "@odoo/hoot";
+import { check, queryAll, queryOne } from "@odoo/hoot-dom";
+import { animationFrame } from "@odoo/hoot-mock";
+import { mountView } from "@web/../tests/web_test_helpers";
+import { mailModels } from "@mail/../tests/mail_test_helpers";
+import { defineProjectModels, projectModels } from "@project/../tests/project_models";
+import { ProjectTask } from "./project_task_model"
```
Comment (pkyriakou) [inline addons/sale_project/static/tests/project_task_list_sale_line.test.js:7]: why the separate import here and not just access it from ``projectModels``?

File: addons/sale_project/static/tests/project_task_list_sale_line.test.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+defineProjectModels();
+
+test("cannot edit sale_line_id when partners are different", async () => {
+    mailModels.ResPartner._records = [
+        { id: 101, name: "Deco Addict" },
+        { id: 102, name: "Azure Interior" },
+        ...mailModels.ResPartner._records,
+    ];
+
+    ProjectTask._records = [
```
Comment (pkyriakou) [inline addons/sale_project/static/tests/project_task_list_sale_line.test.js:19]: we could do this no? 
```suggestion
    projectModels.ProjectTask._records = [
```

File: addons/sale_project/static/tests/project_task_list_sale_line.test.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    await check(".o_list_record_selector input", { root: firstRow });
+    await animationFrame();
+    expect(queryOne("[name=sale_line_id]", { root: firstRow })).not.toHaveClass("o_readonly_modifier");
+    expect(queryOne("[name=sale_line_id]", { root: secondRow })).not.toHaveClass("o_readonly_modifier");
+
+    await check(".o_list_record_selector input", { root: secondRow });
+    await animationFrame();
+    expect(queryOne("[name=sale_line_id]", { root: firstRow })).toHaveClass("o_readonly_modifier");
+    expect(queryOne("[name=sale_line_id]", { root: secondRow })).toHaveClass("o_readonly_modifier");
+});
```
Comment (pkyriakou) [inline addons/sale_project/static/tests/project_task_list_sale_line.test.js:51]: you're missing the string description messages here

File: addons/sale_project/static/tests/project_task_list_sale_line.test.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            <list multi_edit="1" js_class="project_task_list">
+                <field name="partner_id"/>
+                <field name="sale_line_id"/>
+            </list>
+        `,
+    });
+
+    const [firstRow, secondRow] = queryAll(".o_data_row");
+
+    await check(".o_list_record_selector input", { root: firstRow });
```
Comment (pkyriakou) [inline addons/sale_project/static/tests/project_task_list_sale_line.test.js:42]: why check and not click :thinking:

**Review by pkyriakou — COMMENTED** [review]
Thank you for your work @ppr-odoo :100: I left some comments for you :grin:

File: addons/sale_project/static/tests/project_task_model.js (hunk 1 above)
Comment (pkyriakou) [reply addons/sale_project/static/tests/project_task_model.js:21]: also below that, I don't think it's good practice to add ``SaleOrderLine`` as one of the ``projectModels`` in order to define them together. SaleOrderLine is not really a project model - it'd be better to import and define the model "manually" in each test or create a defineSaleModels inside the sales module that defines the sale models (which we'll import in our tests)

File: addons/sale_project/static/tests/project_task_list_sale_line.test.js (hunk 4 above)
Comment (ppr-odoo) [reply addons/sale_project/static/tests/project_task_list_sale_line.test.js:51]: I have done it according to the project module. I believe we should keep everything the same as the project. What do you think? [ref](https://github.com/odoo/odoo/commit/ee27981c3f8aab027c3b2bbe4821f651de6daeff#diff-c22d8c8fed0920e77bf8697f23103d1b4865d17ba91539fc8e6d4ed0d0b0f175R44)

File: addons/sale_project/static/tests/project_task_list_sale_line.test.js (hunk 5 above)
Comment (ppr-odoo) [reply addons/sale_project/static/tests/project_task_list_sale_line.test.js:42]: > why check and not click 🤔

same here

File: addons/sale_project/static/tests/project_task_list_sale_line.test.js (hunk 3 above)
Comment (ppr-odoo) [reply addons/sale_project/static/tests/project_task_list_sale_line.test.js:19]: > we could do this no?

It is possible. however, we are currently accessing it this way in project_enterprise. What is your opinion now?
[ref](https://github.com/odoo/enterprise/commit/f47a8c4f23b81d1a554885767817144dc8377033#diff-381338b2debce4f97fd60ab04b2de784fc7d914595e0d28cbd2e7a76ef055f43R8)

File: addons/sale_project/static/tests/project_task_model.js (hunk 1 above)
Comment (ppr-odoo) [reply addons/sale_project/static/tests/project_task_model.js:21]: > also below that, I don't think it's good practice to add `SaleOrderLine` as one of the `projectModels` in order to define them together. SaleOrderLine is not really a project model - it'd be better to import and define the model "manually" in each test or create a defineSaleModels inside the sales module that defines the sale models (which we'll import in our tests)

I have sent a message to our Vidage server.

**Review by pkyriakou — COMMENTED** [review]
Thank you @ppr-odoo for the updates :100: Nothing to comment on - LGTM :green_circle: :muscle:

**Review by xavierbol — COMMENTED** [review]
LGTM, thanks for your work! 🙂 

robodoo r+

**Comment by xavierbol** [comment]
robodoo r+

**Comment by xavierbol** [comment]
robodoo r+

**Pull request closed without merging**